lib/ansible/modules/crypto/openssh_cert.py

```diff
@@ -0,0 +1,544 @@
+
+            if earliest_time:
+                if not (earliest_time - cert_valid_from).total_seconds == 0.0:
+                    return False
+            if last_time:
+                if not (last_time - cert_valid_to).total_seconds == 0.0:
+                    return False
+
+            if self.valid_at:
+                if cert_valid_from <= self.convert_to_datetime(module, self.valid_at) <= cert_valid_to:
+                    return True
+
+            if earliest_time and last_time:
+                return True
+
+            return False
+
+        if not perms_required:
+            return _check_type() and _check_principals() and _check_validity(module)
+
+        return _check_perms(module) and _check_type() and _check_principals() and _check_validity(module)
+
+    def dump(self):
+
+        """Serialize the object into a dictionary."""
+
+        def filter_keywords(arr, keywords):
+            concated = []
+            string = ""
+            for word in arr:
+                if word in keywords:
+                    concated.append(string)
+                    string = ""
+                string += " " + word
+            concated.append(string)
+            return concated
+
+        def format_cert_info():
+            return filter_keywords(self.cert_info, [
+                "Type:",
+                "Public",
+                "Signing",
+                "Key",
+                "Serial:",
+                "Valid:",
+                "Principals:",
+                "Critical",
+                "Extensions:"])
+
+        result = {
+            'changed': self.changed,
+            'type': self.type,
+            'filename': self.path,
+            'info': format_cert_info(),
+        }
+
+        return result
+
+    def remove(self):
+        """Remove the resource from the filesystem."""
+
+        try:
+            os.remove(self.path)
+            self.changed = True
+        except OSError as exc:
+            if exc.errno != errno.ENOENT:
+                raise CertificateError(exc)
+            else:
+                pass
+        return
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            force=dict(default=False, type=bool),
+            type=dict(required=True, choices=['host', 'user'], type='str'),
+            signing_key=dict(required=True, type='path'),
+            public_key=dict(required=True, type='path'),
+            path=dict(required=True, type='path'),
+            identifier=dict(type='str'),
+            valid_from=dict(required=True, type='str'),
+            valid_to=dict(required=True, type='str'),
+            valid_at=dict(type='str'),
+            principals=dict(type=list),
+            options=dict(type=list),
+        ),
+        supports_check_mode=True,
+        add_file_common_args=True,
+    )
+
+    def isBaseDir(path):
+        base_dir = os.path.dirname(path)
```

Comment:
```suggestion
        base_dir = os.path.dirname(path) or '.'
```
Otherwise, `path == 'file'` (i.e. file in current directory) will result in `fatal: [localhost]: FAILED! => {"changed": false, "msg": "The directory  does not exist or the file is not a directory", "name": ""}`

The same thing goes wrong in `openssh_keypair`, BTW.

Reply:
I created a PR for `openssh_keypair` in #50106.